autoscale/src/main/java/com/sequenceiq/periscope/service/security/RemoteUserDetailsService.java

```diff
@@ -42,38 +44,27 @@
     @Cacheable("userCache")
     @SuppressWarnings("unchecked")
     public PeriscopeUser getDetails(String filterValue, UserFilterField filterField) {
-        HttpHeaders tokenRequestHeaders = new HttpHeaders();
-        tokenRequestHeaders.set("Authorization", getAuthorizationHeader(clientId, clientSecret));
-
-        Map<String, String> tokenResponse = restTemplate.exchange(
-                identityServerUrl + "/oauth/token?grant_type=client_credentials",
-                HttpMethod.POST,
-                new HttpEntity<>(tokenRequestHeaders),
-                Map.class).getBody();
-
-        HttpHeaders scimRequestHeaders = new HttpHeaders();
-        scimRequestHeaders.set("Authorization", "Bearer " + tokenResponse.get("access_token"));
-
-        String scimResponse;
-
+        Client client = clientBuilder.build();
```

Comment:
Client is a heavy wight object, we shall create just a few, we already have a cache for it in api project.